### PR TITLE
Additional Sound Events

### PR DESCRIPTION
**Pull request by peternewell**
https://blueprints.launchpad.net/or/+spec/addtional-steam-locomotive-sound-triggers - Add sound event for gear positions, and adjust injector in AI mode so that sound event only turns on of off with the injector.

### CODE REVIEW

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs [hunk 1]
```diff
@@ -5447,6 +5453,11 @@ private void UpdateFiring(float absSpeedMpS)
                     Injector2IsOn = false;
                     Injector2Fraction = 0.0f;
                     InjectorLockedOut = true;
+                    if (!Injector1SoundIsOn) // turn injector sound on only when injector starts
+                    {
+                        Injector1SoundIsOn = true;
+                        SignalEvent(Event.WaterInjector1On);
+                    }
```
Comment (YoRyan) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs:5460]: I notice you repeat this logic many times. Consider factoring it out into a helper method?

**Comment by YoRyan** [comment]
Hi Peter,

Code looks good. Please also add documentation of the new triggers to the manual.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs (hunk 1 above)
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs:5460]: Unfortunately I am a "self taught" coder, so always happy to have examples of better ways to code stuff and achieve the same outcome.

Thanks

Comment (YoRyan) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs]: See 7952b1944047a2197044f4b9707af1579a0431e5.

**Comment by YoRyan** [comment]
Approved with a manual update (courtesy of Peter) and some minor tweaks. Thanks again for your contributions.

**Pull request merged**